core/src/main/scala/com/microsoft/azure/synapse/ml/io/http/RESTHelpers.scala

```diff
@@ -89,7 +89,13 @@ object RESTHelpers {
   }
 
   def parseResult(result: CloseableHttpResponse): String = {
-    IOUtils.toString(result.getEntity.getContent, "utf-8")
+    try {
+      IOUtils.toString(result.getEntity.getContent, "utf-8")
+    }
+    catch{
+      case ex: Exception =>
+        "{}"
+    }
```

Comment:
I dont think this is really the right place to be modifying what comes back. This API is intended to be a faithful representation of what is returned. If theres an exception, a user would want to know what the exception is. If there was no content returned a user would expect the empty string as opposed to an empty json object. Perhaps putting a check for an empty string higher up the stack might make sense, but this low in the stack isnt the right place for this fix. What is the key exception you are running into that prompted this?

Reply:
The exception that I encountered is in "What changes are proposed in this PR". Here is the exception: 

"spray.json.JsonParser$ParsingException: Unexpected end-of-input at input index 0 (line 1, position 1), expected JSON Value:

^"

Reply:
I have moved the check to one frame up in the stack. Can you please take a look?